src/Runner.Common/RunServer.cs

```diff
@@ -70,7 +70,43 @@ private void CheckConnection()
         public Task<AgentJobRequestMessage> GetJobMessageAsync(string id)
```

Comment:
do we need to pass the cancellation token from the caller?

Reply:
+1